packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -215,49 +222,159 @@ class RealmDictionaryTests {
+            "A" to DICTIONARY_OBJECT_VALUES[0]
+        ).also { dictionary ->
+            val container = RealmDictionaryContainer()
+                .apply { nullableObjectDictionaryField = dictionary }
+
+            realm.writeBlocking {
+                val managedContainer = copyToRealm(container)
+                val managedDictionary = managedContainer.nullableObjectDictionaryField
+                assertEquals(dictionary.size, managedDictionary.size)
+                val expected = assertNotNull(dictionary["A"])
+                val actual = assertNotNull(managedDictionary["A"])
+                assertEquals(expected.stringField, actual.stringField)
             }
+
+            // Repeat outside transaction
+            realm.query<RealmDictionaryContainer>()
+                .first()
+                .find {
+                    val managedDictionary = assertNotNull(it).nullableObjectDictionaryField
+                    assertEquals(dictionary.size, managedDictionary.size)
+                    val expected = assertNotNull(dictionary["A"])
+                    val actual = assertNotNull(managedDictionary["A"])
+                    assertEquals(expected.stringField, actual.stringField)
+                }
+        }
+
+        // Cleanup between assertions for convenience
+        realm.writeBlocking {
+            delete(query<RealmDictionaryContainer>())
+        }
+
+        // Test with a dictionary of RealmAny containing a primitive value
+        realmDictionaryOf(
+            "A" to REALM_ANY_PRIMITIVE_VALUES[0]
+        ).also { dictionary ->
+            val container = RealmDictionaryContainer()
+                .apply { nullableRealmAnyDictionaryField = dictionary }
+
+            realm.writeBlocking {
+                val managedContainer = copyToRealm(container)
+                val managedDictionary = managedContainer.nullableRealmAnyDictionaryField
+                assertEquals(dictionary.size, managedDictionary.size)
+                val expected = assertNotNull(dictionary["A"])
+                val actual = assertNotNull(managedDictionary["A"])
+                assertEquals(expected, actual)
+            }
+
+            // Repeat outside transaction
+            realm.query<RealmDictionaryContainer>()
+                .first()
+                .find {
+                    val managedDictionary = assertNotNull(it).nullableRealmAnyDictionaryField
+                    assertEquals(dictionary.size, managedDictionary.size)
+                    val expected = assertNotNull(dictionary["A"])
+                    val actual = assertNotNull(managedDictionary["A"])
+                    assertEquals(expected, actual)
+                }
+        }
+
+        // Cleanup between assertions for convenience
+        realm.writeBlocking {
+            delete(query<RealmDictionaryContainer>())
+        }
+
+        // Test with a dictionary of RealmAny containing an object
+        realmDictionaryOf<RealmAny?>(
```

Comment:
I would move this into its own test. This test is bit too big and tests too many things IMO. Same in the other tests

Reply:
I'll split both getter and setter sets into the different cases.